### PR TITLE
Fixed mendel-mocha variational test issue

### PR DESCRIPTION
**Pull request by stephanwlee**
Bug in getEntriesByGlob filtered out the entries in the
variational directories.

### CODE REVIEW

File: packages/mendel-pipeline/src/registry/client.js [hunk 1]
```diff
@@ -110,7 +104,11 @@ class MendelOutletRegistry {
             return positives.some(g => g.match(id)) &&
                 negatives.every(g => g.match(id));
         })
-        .map(id => this.getEntry(id));
+        .map(id => this.getEntry(id))
+        .filter(({variation}) => {
+            return !excludeVariational ||
+                variation === this._options.baseConfig.dir;
+        });
```
Comment (irae) [inline packages/mendel-pipeline/src/registry/client.js]: Why do we exclude variational in here? For instance, what about a variation that introduces a bundle? The _entrance_ would only exist in the new bundle. I was under the impression that picking up from all variation dirs was the proper behavior. It might be breaking something, but Not in full example, at least.

File: packages/mendel-mocha-runner/index.js [hunk 2]
```diff
@@ -36,7 +36,7 @@ function MendelRunner(filePaths, options={}) {
 
         const mocha = new Mocha(options);
 
-        const entries = client.registry.getEntriesByGlob(filePaths);
+        const entries = client.registry.getEntriesByGlob(filePaths, false);
```
Comment (irae) [inline packages/mendel-mocha-runner/index.js]: See suggestion about not filtering. But if we keep the filtering, can we have meaningful options instead, even if this is the only config so far?

```js
// usage
client.registry.getEntriesByGlob(filePaths,{includeVariations:true});
// implementation
getEntriesByGlob(globStrings, options = {includeVariations:false}) {/*...*/}
```

File: packages/mendel-pipeline/src/registry/client.js (hunk 1 above)
Comment (stephanwlee) [inline packages/mendel-pipeline/src/registry/client.js]: >  I was under the impression that picking up from all variation dirs was the proper behavior. It might be breaking something, but Not in full example, at least.

Not true. 
https://github.com/yahoo/mendel/blob/master/examples/full-example/.mendelrc#L135
https://github.com/yahoo/mendel/blob/e46e7bf1c67d28c12f7566432e58a4406bbb9864/packages/mendel-pipeline/src/bundles/default-generator.js#L10

Imposing that you create a file in base variation for "entries" is not a bad one. Btw. `walk` will take care of variations. 
https://github.com/yahoo/mendel/blob/e46e7bf1c67d28c12f7566432e58a4406bbb9864/packages/mendel-pipeline/src/registry/client.js#L124

In summary, there may be no harm in walking in variational dir too but there is a strong chance it will do redundant walking.

Comment (stephanwlee) [inline packages/mendel-pipeline/src/registry/client.js]: Will remove the filter :\
